tsdb/head.go

```diff
@@ -1853,6 +1859,33 @@ func newStripeSeries(stripeSize int, seriesCallback SeriesLifecycleCallback) *st
 	return s
 }
 
+// reset should reset the stripeSeries and appropriately trigger the associated hooks.
+// note: This method simply resets the stripeSeries struct to its initial state without
+// performing any operations on the chunks.
+func (s *stripeSeries) reset() int {
+	deletedFromPrevStripe := 0
+	totalSeries := 0
+	for i := 0; i < s.size; i++ {
```

Comment:
Could this code be shared with `gc()` around line 1960?
I.e. via some refactoring.  To avoid diverging bugs going forward.

Reply:
make sense. I created a `s.Iter` method that is now shared btw both methods.